kpi/deployment_backends/mixin.py

```diff
@@ -18,11 +18,15 @@ def connect_deployment(self, **kwargs):
 
     def deploy(self, backend=False, active=True):
         '''this method could be called "deploy_latest_version()".'''
-        if not self.has_deployment:
-            self.connect_deployment(backend=backend, active=active)
+
+        if self.can_be_deployed:
+            if not self.has_deployment:
+                self.connect_deployment(backend=backend, active=active)
+            else:
+                self.deployment.redeploy(active=active)
+            self._mark_latest_version_as_deployed()
         else:
-            self.deployment.redeploy(active=active)
-        self._mark_latest_version_as_deployed()
+            raise BadAssetTypeException("A form template can't be deployed")
```

Comment:
This message should probably match whatever you choose for https://github.com/kobotoolbox/kpi/pull/1806/files#diff-43986855ae1015d5975b107cf1ee9869R691 (which I have suggested changing since only `survey`s can be deployed)